CHAOS-5501 Linux JVM trigger gc (#6790)
harness/developer-hub: docs/chaos-engineering/chaos-faults/linux/linux-jvm-trigger-gc.md

---
id: linux-jvm-trigger-gc
title: Linux JVM trigger gc
---

import Ossupport from './shared/note-supported-os.md'
import FaultPermissions from './shared/fault-permissions.md'

Linux JVM trigger gc triggers the garbage collector on a specific process in Java that causes unused (or out of scope) objects, variables and so on to be garbage collected and recycled, thereby freeing up memory space. It is used to determine the performance and resilience of the application (or services) running on Linux machines.

:::tip
JVM chaos faults use the [Byteman utility](https://byteman.jboss.org/) to inject chaos faults into the JVM.
:::

![Linux JVM trigger gc](./static/images/linux-jvm-trigger-gc.png)

## Use cases
Linux JVM trigger gc:
- Determines how the application behaves when memory space is freed up randomly for a brief period.
- Determines how efficiently an application recovers and returns to normalcy.

<Ossupport />

<FaultPermissions />

### Mandatory tunables
<table>
  <tr>
    <th> Tunable </th>
    <th> Description </th>
    <th> Notes </th>
  </tr>
  <tr>

>     <th> Pid </th>

Reviewer: Please check in all the other places too, `<th>` should only be used in heading cells.
```suggestion
    <td> Pid </td>
```

Author: The changes to html elements is intentional, the tunables are highlighted this way and easy to identify them.

    <td> The process Id that Byteman uses to target the service. This is mutually exclusive with <b>Startup command</b>. </td>
    <td> For example, <code>6429</code>. For more information, go to <a href="#pid"> process Ids</a>.</td>
  </tr>
  <tr>
    <th> Startup command </th>

>     <td> The path to the JAR file or the Java application. This is mutually exclusive with <b>Pid</b>.</td>

Reviewer: ```suggestion
    <td> The command used to start the Java process. A substring match is performed for the given value for all the processes. This is mutually exclusive with <b>Pid</b>.</td>
```

    <td> For example, <code>/usr/local/bin/pet-clinic.jar</code>. For more information, go to <a href= "#startup-command">startup command.</a></td>
  </tr>
</table>

### Optional tunables
<table>
  <tr>
    <th> Tunable </th>
    <th> Description </th>
    <th> Notes </th>
  </tr>
  <tr>
    <th> Duration </th>
    <td> Duration through which chaos is injected into the target resource. Should be provided in <code>[numeric-hours]h[numeric-minutes]m[numeric-seconds]s</code> format. </td>
    <td> Default: <code>30s</code>. Examples: <code>1m25s</code>, <code>1h3m2s</code>, <code>1h3s</code>. For more information, go to <a href="/docs/chaos-engineering/chaos-faults/common-tunables-for-all-faults/#duration-of-the-chaos"> duration of the chaos.</a></td>
  </tr>
  <tr>
    <th> Port </th>

>     <td> Port used to Byteman to start the Byteman agent. </td>

Reviewer: ```suggestion
    <td> Port to be used by the Byteman agent. </td>
```

>     <td> For example, <code>9091</code>. </td>

Reviewer: ```suggestion
    <td> Default: <code>9091</code>. </td>
```

  </tr>
  <tr>

>     <th> Ramp time </th>

Reviewer: Please write all of these in camelCase, that's how the input is provided in the manifest too.

Check in other places too.
```suggestion
    <th> rampTime </th>
```

    <td> Period to wait before and after injecting chaos. Should be provided in <code>[numeric-hours]h[numeric-minutes]m[numeric-seconds]s</code> format. </td>
    <td> Default: <code>0s</code>. Examples: <code>1m25s</code>, <code>1h3m2s</code>, <code>1h3s</code>. For more information, go to <a href= "/docs/chaos-engineering/chaos-faults/common-tunables-for-all-faults#ramp-time">ramp time.</a></td>
  </tr>
  <tr>
    <td>  </td>
    <td>  </td>
    <td>  </td>

>   </tr>

Reviewer: We can remove this?

Reviewer: Please check in other faults too.
